src/Analyzer.PowerShellRuleEngine/PowerShellRuleEngine.cs

```diff
@@ -0,0 +1,115 @@
+        public static IEnumerable<IEvaluation> EvaluateRules(string templateFilePath)
+        {
+            var powerShell = System.Management.Automation.PowerShell.Create();
+
+            powerShell.Streams.Error.DataAdded += (sender, e) => HandleDataAddedInStreams(powerShell.Streams.Error[e.Index], ConsoleColor.Red);
+            powerShell.Streams.Warning.DataAdded += (sender, e) => HandleDataAddedInStreams(powerShell.Streams.Warning[e.Index], ConsoleColor.Yellow);
+            powerShell.Streams.Information.DataAdded += (sender, e) => HandleDataAddedInStreams(powerShell.Streams.Information[e.Index]);
+            powerShell.Streams.Verbose.DataAdded += (sender, e) => HandleDataAddedInStreams(powerShell.Streams.Verbose[e.Index]);
+            powerShell.Streams.Debug.DataAdded += (sender, e) => HandleDataAddedInStreams(powerShell.Streams.Debug[e.Index]);
+
+            powerShell.Commands.AddCommand("Set-ExecutionPolicy")
+                .AddParameter("Scope", "Process") // Affects only the current PowerShell session
+                .AddParameter("ExecutionPolicy", "Unrestricted");
+            powerShell.AddStatement();
+
+            powerShell.Commands.AddCommand("Import-Module")
+                .AddParameter("Name", @".\TTK\arm-ttk.psd1"); // arm-ttk is added to the needed project's bins directories in build time 
+            powerShell.AddStatement();
+
+            powerShell.Commands.AddCommand("Test-AzTemplate")
+                .AddParameter("Test", "deploymentTemplate")
+                .AddParameter("TemplatePath", templateFilePath);
+
+            var executionResults = powerShell.Invoke();
+
+            var evaluations = new List<PowerShellRuleEvaluation>();
+
+            foreach (dynamic executionResult in executionResults)
+            {
+                var uniqueErrors = new Dictionary<string, SortedSet<int>>(); // Maps error messages to a sorted set of line numbers
+
+                foreach (dynamic warning in executionResult.Warnings)
+                {
+                    AddErrorToDictionary(warning, ref uniqueErrors);
+                }
+
+                foreach (dynamic error in executionResult.Errors)
+                {
+                    AddErrorToDictionary(error, ref uniqueErrors);
+                }
+
+                foreach (KeyValuePair<string, SortedSet<int>> uniqueError in uniqueErrors)
+                {
+                    var evaluationResults = new List<PowerShellRuleResult>();
+                    foreach (int lineNumber in uniqueError.Value)
+                    {
+                        var evaluationResult = new PowerShellRuleResult(false, lineNumber);
+                        evaluationResults.Add(evaluationResult);
+                    }
+                    var evaluation = new PowerShellRuleEvaluation(executionResult.Name, uniqueError.Key, false, evaluationResults);
+                    evaluations.Add(evaluation);
```

Comment:
Suggestion: instead of assigning the new evaluation and result to variables and then adding them to the list, create the new objects directly in the calls to `Add()`.